### PR TITLE
refactor: return `204` instead of `200` for `PUT /authenticator/:id`

### PR DESCRIPTION
**Pull request by sstrigler**
Fixes [EMQX-7982](https://emqx.atlassian.net/browse/EMQX-7982)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
- [x] ~For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)~
- [x] For internal contributor: there is a jira ticket to track this change
- [x] ~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~
- [x] ~In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section~
- [x] Sync with dashboard team

### CODE REVIEW

File: changes/v5.0.12-zh.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 - 通过 `node.global_gc_interval = disabled` 来禁用全局垃圾回收 [#9418](https://github.com/emqx/emqx/pull/9418)。
 
+- [FIXME] - Return `204` instead of `200` for `PUT /authenticator/:id` [#9434](https://github.com/emqx/emqx/pull/9434/).
```
Comment (lafirest) [inline changes/v5.0.12-zh.md:7]: 现在，`PUT /authenticator/:id` 将会返回 `204` 而不再是 `200` [#9434](https://github.com/emqx/emqx/pull/9434/)。

**Review by qzhuyan — APPROVED** [review]

File: apps/emqx_authn/src/emqx_authn_api.erl [hunk 2]
```diff
@@ -1006,11 +1006,10 @@ update_authenticator(ConfKeyPath, ChainName, AuthenticatorID, Config) ->
         )
     of
         {ok, #{
-            post_config_update := #{emqx_authentication := #{id := ID}},
-            raw_config := AuthenticatorsConfig
+            post_config_update := #{emqx_authentication := #{id := _ID}},
+            raw_config := _AuthenticatorsConfig
         }} ->
-            {ok, AuthenticatorConfig} = find_config(ID, AuthenticatorsConfig),
-            {200, maps:put(id, ID, convert_certs(fill_defaults(AuthenticatorConfig)))};
```
Comment (zmstone) [inline apps/emqx_authn/src/emqx_authn_api.erl:1013]: I'm afraid this has to be reverted.

Comment (zmstone) [reply apps/emqx_authn/src/emqx_authn_api.erl:1013]: this is not an echo, the `convert_certs` will store files to disk, and return stored paths.
and `fill_defaults` will fill default values for the missing fields in the request.

Comment (sstrigler) [reply apps/emqx_authn/src/emqx_authn_api.erl:1013]: I've been reading through the code a bit now. `convert_certs` does have side-effects, it *reads* `pem` files and checks if they are valid. But it doesn't look like it's creating anything anywhere. 

But `update_config` might trigger creating of ssl related files. Couldn't quite trace it back but there is `emqx_authentication_config:pre_config_update` which later calls some other `convert_certs` within its own module. And this one, deep down the line does a call to `save_pem_file` eventually. 

I just assume that through some callbacks `update_config` does call `emqx_authentication_config:pre_config_update` and as such create those `pem` file(s). Means it should happen with or without that call to that *local* `convert_certs`. 

It still might be necessary to do call those checks to trigger a crash just to indicate there's something wrong, but in that case a proper error handling would we preferable. And it should already happen within `update_config` I'd think.

**Pull request merged**